lib/handlers/quote/quote.ts

```diff
@@ -416,6 +424,14 @@ export class QuoteHandler extends APIGLambdaHandler<
       case 'exactOut':
         amount = CurrencyAmount.fromRawAmount(currencyOut, JSBI.BigInt(amountRaw))
 
+        if (!amount.greaterThan(CurrencyAmount.fromRawAmount(currencyIn, JSBI.BigInt(0)))) {
+          return {
+            statusCode: 400,
```

Comment:
is `400` recognized by upstream? i see we return `404` for no liquidity

Reply:
yea URA will throw non-404 errors for [retryability](https://github.com/Uniswap/backend/blob/d7ef4b2af406ece6faf4311d3ba898e2877dc855/packages/services/unified-routing-api/lib/providers/quoters/RoutingApiQuoter.ts#L117). 

we have another [quote test with amount negative](https://github.com/Uniswap/routing-api/blob/c8b8fdca4399643083c007943c421768cacf5d6e/test/mocha/e2e/quote.test.ts#L2573), that asserts 400 as well.